pkg/health/checker/checker.go

```diff
@@ -443,12 +443,16 @@ func (h shadowTrafficHealthChecker) WatchService(
 					case resultCh <- healthResults:
 					}
 				case nodeIDs = <-nodeIDsCh:
-					for _, nodeID := range nodeIDs {
-						healthResults[nodeID] = health.Result{
-							ID:      types.PodID(serviceID),
-							Node:    nodeID,
-							Service: serviceID,
-							Status:  health.Passing,
+					// app is considered always healthy when
+					// status port is not set
+					if status.Port == 0 {
```

Comment:
I wonder if this code would be better factored if the `status.Port` check lived in the loop that iterates over nodeIDs instead of in this select loop.

Reply:
The loop over nodeIDs for setting healthResults only needs to be done if the app is always healthy, in the the default case, nodeIDs still get set to the new values. The monitoring connection is reset every 10 minutes and will be using the latest values of nodeIDs when making a new connection.

Reply:
I was expressing a desire to have this knowledge (status.Port == 0 -> always healthy) be encapsulated at a lower layer so it doesn't have to appear twice in the health checker. For example, if the healthClient captured this knowledge and the health checker was naive about it, we wouldn't have had yesterday's bug.

This is a major refactor that I can't reasonably suggest doing without a test suite, so don't worry about it for this PR.